File: backend/portal/urls/organisation.py
from django.urls import path

from ..views.organisation import (
    organisation_create,
    organisation_leave,
)


urlpatterns = [
    path(
        "teach/onboarding-organisation/",
        organisation_create,
        name="organisation_create",
    ),
    path(
        "teach/leave-organisation/",
        organisation_leave,
        name="organisation_leave",
    ),
]

File: frontend/src/app/api/organisation.ts
import api from './api';
import { MoveClassDataProps } from './teacher/dashboard';

const organisationApi = api.injectEndpoints({
  endpoints: (build) => ({
    createOrganisation: build.mutation<void, {
      name: string;
      postcode: string;
      country: string;
    }>({
      query: (body) => ({
        url: 'teach/onboarding-organisation/',
        method: 'POST',
        body,
        headers: {
          'Content-Type': 'application/x-www-form-urlencoded'
        }
      })
    }),
    leaveOrganisation: build.mutation<MoveClassDataProps, void | object
    >({
      query: (body) => ({
        url: 'teach/leave-organisation/',
        method: 'POST',
        body,
        headers: {
          'Content-Type': 'application/x-www-form-urlencoded'
        }
      })
    })
  })
});

export default organisationApi;
export const {
  useCreateOrganisationMutation,
  useLeaveOrganisationMutation
} = organisationApi;
